pojo/src/test/java/com/spotify/hamcrest/pojo/IsPojoTest.java

```diff
@@ -224,16 +228,11 @@ public void testTypeSafeMismatch() throws Exception {
 
   @Test
   public void testMultipleIdenticalMatches() throws Exception {
-    final IsPojo<SomeClass> sut = pojo(SomeClass.class)
+    expectedException.expect(both(isA(IllegalArgumentException.class))
+        .and(hasProperty("message", startsWith("Multiple entries with same key: "))));
```

Comment:
I think it would be nice to inform user this with a more detailed error message than that from Guava. Some new users to `hamcrest` might not know to use `allOf()` for example. Maybe something like

```
Multiple wheres found for method <method-name>. This is a not supported operation. Instead of using multiple wheres, the use of Matchers.allOf or Matchers.both is encouraged. See the hamcrest documentation for more details
```